CRM/Logging/ReportDetail.php

```diff
@@ -218,25 +221,68 @@ protected function convertDiffsToRows() {
           $to = implode(', ', array_filter($tos));
         }
 
+        $cfArray = [];
         $tableDAOClass = CRM_Core_DAO_AllCoreTables::getClassForTable($table);
+        $fkClassName = NULL;
         if (!empty($tableDAOClass)) {
           $tableDAOFields = (new $tableDAOClass())->fields();
           // If this field is a foreign key, then we can later use the foreign
           // class to translate the id into something more useful for display.
           $fkClassName = $tableDAOFields[$field]['FKClassName'] ?? NULL;
         }
+        else {
+          // Since this table didn't match a core table, check if it's a custom field.
+          $cfResultSet = \Civi\Api4\CustomField::get()
+            ->addSelect('id', 'data_type', 'fk_entity', 'option_group_id')
+            ->addWhere('column_name', '=', $field)
+            ->setLimit(1)
+            ->execute();
+          if (is_array($cfResultSet->first())) {
+            $cfArray = $cfResultSet->first();
+          }
```

Comment:
This api call should *not* include permission checks since this is an internal function.
But also... we shouldn't need to use the api & hit the database for this since we have more efficient getters for custom fields (`CRM_Core_BAO_CustomField::getField()` and custom groups (`CRM_Core_BAO_CustomGroup::getAll()`).

But also... there's something odd about this code - why is it matching on column name only? That's not specific enough. Different custom fields could have the same column name in different groups. E.g. you could have `MyGroupOne.my_field_column` and also `MyGroupTwo.my_field_column` and this code would not be able to tell the difference.

Reply:
This is column_name though not `name` name. It is unique and looks like `number_of_cats_27` with the field id at the end. But yes good point there might be a more efficient way to get it.

Reply:
Ok, I think that id-appendage is done at the form layer & we can't count on it definitely being there, but if the norm is that column names are unique then I guess it's safe enough for us to treat them as unique here. If somebody creates custom fields via API & accidentally makes them non-unique, that's on them.
In that case, the efficient way to get a custom field by column name would be something like:
```
foreach (CRM_Core_BAO_CustomGroup::getAll() as $customGroup) {
  foreach ($customGroup['fields'] as $customField) {
    if ($customField['column_name'] === $field) {
      break 2;
    }
  }
}
```

Reply:
If the table_name were included as part of the comparison, then it would definitely be unique. The table name is available here already as `$table`.

Reply:
Ah, yes, perfect. That's what was missing from the api call. And that makes the code much simpler. It can be:
```php
$customGroup = CRM_Core_BAO_CustomGroup::getGroup(['table_name' => $table]);
foreach ($customGroup['fields'] as $customField) {
  if ($customField['column_name'] === $field) {
    break;
  }
}
```

Reply:
@justinmosier see ^ for a streamlined version